src/crypto_helper/computations.rs

```diff
@@ -92,3 +94,41 @@ pub fn process_zlib(input: &ZlibInput) -> Result<Vec<u8>, String> {
         }
     }
 }
+
+pub fn process_argon2(input: &Argon2Input) -> Result<Vec<u8>, String> {
+    let config = base64::Config::new(base64::CharacterSet::Standard, false);
+    match &input.action {
+        Argon2Action::Hash(hash_action) => {
+            let argon2ctx = argon2::Argon2::new(
+                hash_action.variant.into(),
+                hash_action.version.into(),
+                hash_action.into(),
+            );
+
+            let salt = if hash_action.salt.is_empty() {
+                password_hash::SaltString::generate(rand::thread_rng())
+            } else {
+                password_hash::SaltString::from_b64(&base64::encode_config(&hash_action.salt, config))
+                    .map_err(|e| e.to_string())?
+            };
+
+            let hash = argon2ctx
+                .hash_password(&input.data, salt.as_salt())
+                .map_err(|err| err.to_string())?.to_string();
+
+            Ok(hash.into_bytes())
+        }
+        Argon2Action::Verify(data) => {
+            let b64 = base64::encode_config(data, config);
+
+            let hash = argon2::PasswordHash::parse(&b64, argon2::password_hash::Encoding::B64)
+                .map_err(|e| e.to_string())?;
+
+            if argon2::Argon2::default().verify_password(data, &hash).is_ok() {
+                Ok(vec![1])
+            } else {
+                Ok(vec![0])
+            }
+        }
```

Comment:
The password verification algorithm is wrong. To verify the password, you need two items: the password itself and the argon2 hash of this password

Reply:
Yeah, it's not ready yet :(

Reply:
Password hashing looks good to me. so, password verification is the only thing left